app/knight.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.knight = knight
+
+    def preparation(self) -> None:
+        self.name = self.knight["name"]
+        self.power = (self.knight["power"]
+                      + (self.knight["weapon"].get("power", 0)
+                         if self.knight["weapon"] else 0)
+                      + (self.knight["potion"]["effect"].get("power", 0)
+                         if self.knight["potion"] else 0))
+        self.hp = (self.knight["hp"]
+                   + (self.knight["potion"]["effect"].get("hp", 0)
+                      if self.knight["potion"] else 0))
+        self.protection = ((sum(protec["protection"]
+                                for protec in self.knight["armour"])
+                            if self.knight["armour"] else 0)
+                           + (self.knight["potion"]["effect"].get("protection"
+                                                                  , 0)
+                              if self.knight["potion"] else 0))

```

Comment:
Same issue as in line 14. Consider creating helper functions to avoid code repetition. This is a code quality issue.